### ISSUE TITLE
Codes in prompt_segmenting.py

### ISSUE THREAD

**Issue by jsxzs** [Jumpat/SegAnyGAussians#43]
Hi. I am a little bit confused about your code in `prompt_segmenting.py`:
```python
# Filter out the points confirmed to be negative

final_mask = point_mask.float().detach().clone().unsqueeze(-1)
final_mask.requires_grad = True

background = torch.zeros(final_mask.shape[0], 3, device = 'cuda')
rendered_mask_pkg = gaussian_renderer.render_mask(cameras[ref_img_camera_id], feature_gaussians, pipeline.extract(args), background, precomputed_mask=final_mask)

tmp_target_mask = torch.tensor(origin_ref_mask, device=rendered_mask_pkg['mask'].device)
tmp_target_mask = torch.nn.functional.interpolate(tmp_target_mask.unsqueeze(0).unsqueeze(0).float(), size=rendered_mask_pkg['mask'].shape[-2:] , mode='bilinear').squeeze(0)
tmp_target_mask[tmp_target_mask > 0.5] = 1
tmp_target_mask[tmp_target_mask != 1] = 0

loss = 30*torch.pow(tmp_target_mask - rendered_mask_pkg['mask'], 2).sum()
loss.backward()

grad_score = final_mask.grad.clone()
final_mask = final_mask - grad_score
final_mask[final_mask < 0] = 0
final_mask[final_mask != 0] = 1
final_mask *= point_mask.unsqueeze(-1)
```
Could you please explain the code here? Why you compute the loss and conduct backward() to get the final_mask?
Thank you.

**Comment by Jumpat** [comment]
Hello, sorry I missed the notification. This operation project a 2D mask to the corresponding 3D Gaussians by gradient descent. This process is similar to the inverse rendering in [SA3D](https://github.com/Jumpat/SegmentAnythingin3D).

**Comment by jsxzs** [comment]
Thank you so much.
